src/pages/registration/contributors_tab/Contributors.tsx

```diff
@@ -0,0 +1,178 @@
+  contributorRole?: ContributorRole;
+}
+
+export const Contributors = ({ contributorRole = ContributorRole.CREATOR, push, replace }: ContributorsProps) => {
+  const { t } = useTranslation('registration');
+  const dispatch = useDispatch();
+  const { values, setFieldValue, setFieldTouched } = useFormikContext<Registration>();
+  const {
+    entityDescription: { contributors },
+  } = values;
+  const [openContributorModal, setOpenContributorModal] = useState(false);
+  const [unverifiedContributor, setUnverifiedContributor] = useState<UnverifiedContributor | null>(null);
+  const isMobile = useIsMobile();
+
+  const relevantContributors = contributors.filter((contributor) => contributor.role === contributorRole);
+  const otherContributors = contributors.filter((contributor) => contributor.role !== contributorRole);
+
+  const handleOnRemove = (indexToRemove: number) => {
+    const nextRelevantContributors = relevantContributors
+      .filter((_, index) => index !== indexToRemove)
+      .map((contributor, index) => ({ ...contributor, sequence: index + 1 }));
+    const nextContributors = [...nextRelevantContributors, ...otherContributors];
+    setFieldValue(ContributorFieldNames.CONTRIBUTORS, nextContributors);
+
+    if (nextContributors.length === 0) {
+      // Ensure field is set to touched even if it's empty
+      setFieldTouched(ContributorFieldNames.CONTRIBUTORS);
+    }
+  };
+
+  const handleMoveContributor = (newSequence: number, oldSequence: number) => {
+    const oldIndex = relevantContributors.findIndex((c) => c.sequence === oldSequence);
+    const minNewIndex = 0;
+    const maxNewIndex = relevantContributors.length - 1;
+
+    const boundedNewIndex =
+      newSequence > relevantContributors.length ? maxNewIndex : newSequence < 0 ? minNewIndex : null;
+    const newIndex = boundedNewIndex
+      ? boundedNewIndex
+      : Math.min(
+          maxNewIndex,
+          Math.max(
+            minNewIndex,
+            relevantContributors.findIndex((c) => c.sequence === newSequence)
+          )
+        );
+
+    const reorderedContributors = move(relevantContributors, oldIndex, newIndex) as Contributor[];
+    // Ensure incrementing sequence values
+    const newContributors = reorderedContributors.map((contributor, index) => ({
+      ...contributor,
+      sequence: index + 1,
+    }));
+    setFieldValue(ContributorFieldNames.CONTRIBUTORS, [...otherContributors, ...newContributors]);
+  };
+
+  const handleOpenContributorModal = (unverifiedAuthor: UnverifiedContributor) => {
+    setUnverifiedContributor(unverifiedAuthor);
+    setOpenContributorModal(true);
+  };
+
+  const onAuthorSelected = (authority: Authority) => {
+    if (relevantContributors.some((author) => author.identity.id === authority.id)) {
+      dispatch(setNotification(t('contributors.author_already_added'), NotificationVariant.Info));
+      return;
+    }
+
+    const identity: Identity = {
+      type: BackendTypeNames.IDENTITY,
```

Comment:
denne trengs ikke etter ny datamodell. skal vi skrive en TODO eller bare gå igjennom alle types når vi begynner på datamodell-endringer?

Reply:
ser det heter `nameType` i den nye datamodellen. Det er kanskje det det skal byttes ut med?

Reply:
tenker vi bare må gå gjennom alt når vi skal bumpe. there will be pain...